File: solutions/python/unique-characters.py
def uniqueChars(str):
    char_bools = [False] * 256
    
    for character in str:
        char_code = ord(character)
        if char_bools[char_code] == True:
            return True
        else:
            char_bools[char_code] = True

    return False


print(uniqueChars('abcd')) # --> False
print(uniqueChars('abbc')) # --> True

File: solutions/javascript/unique-characters.js
var uniqueChars = function(str) {
  var existsBools = [];
  
  for (var i = 0; i < 256; i++) {
    existsBools.push(false);
  }

  for (var i = 0; i < str.length; i++) {
    var boolIndex = str.charCodeAt(i);
    if (existsBools[boolIndex]) {
      return false;
    } else {
      existsBools[boolIndex] = true
    }
  }

  return true;
};

console.log(uniqueChars('abcde')) // --> true
console.log(uniqueChars('abbc')) // --> false

